src/OctoshiftCLI.IntegrationTests/TestHelper.cs

```diff
@@ -0,0 +1,564 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Newtonsoft.Json.Linq;
+using Xunit.Abstractions;
+
+namespace OctoshiftCLI.IntegrationTests
+{
+    public class TestHelper
+    {
+        private readonly ITestOutputHelper _output;
+        private readonly AdoApi _adoApi;
+        private readonly GithubApi _githubApi;
+        private readonly AdoClient _adoClient;
+        private readonly GithubClient _githubClient;
+
+        public TestHelper(ITestOutputHelper output, AdoApi adoApi, GithubApi githubApi, AdoClient adoClient, GithubClient githubClient)
+        {
+            _output = output;
+            _adoApi = adoApi;
+            _githubApi = githubApi;
+            _adoClient = adoClient;
+            _githubClient = githubClient;
+        }
+
+        public TestHelper(ITestOutputHelper output, GithubApi githubTargetApi, GithubClient githubClient)
+        {
+            _output = output;
+            _githubApi = githubTargetApi;
+            _githubClient = githubClient;
+        }
+
+        public async Task ResetAdoTestEnvironment(string adoOrg)
+        {
+            var teamProjects = await _adoApi.GetTeamProjects(adoOrg);
+
+            _output.WriteLine($"Found {teamProjects.Count()} Team Projects");
+
+            foreach (var teamProject in teamProjects.Where(x => x != "service-connection-project-do-not-delete"))
+            {
+                _output.WriteLine($"Deleting Team Project: {adoOrg}\\{teamProject}...");
+                var teamProjectId = await _adoApi.GetTeamProjectId(adoOrg, teamProject);
+                var operationId = await DeleteTeamProject(adoOrg, teamProjectId);
+
+                while (await GetOperationStatus(adoOrg, operationId) is "notSet" or "queued" or "inProgress")
```

Comment:
nit: Def. not a biggie but I'd create constants for these strings (`NOT_SET`, `QUEUED`, `IN_PROGRESS`) since they are being used in other methods too.

Reply:
done